raphtory/src/db/api/view/time.rs

```diff
@@ -203,6 +215,20 @@ impl<'graph, V: OneHopFilter<'graph> + 'graph + InternalTimeOps<'graph>> TimeOps
         self.internal_window(time, time.map(|t| t.saturating_add(1)))
     }
 
+    fn snapshot_at<T: IntoTime>(&self, time: T) -> Self::WindowedViewType {
+        match self.current_filter().graph_type() {
+            GraphType::EventGraph => self.before(time.into_time() + 1),
+            GraphType::PersistentGraph => self.at(time),
+        }
+    }
+
+    fn snapshot_latest(&self) -> Self::WindowedViewType {
```

Comment:
Possibly worth for the event graph just returning self? Will be faster this way as there wont be a window